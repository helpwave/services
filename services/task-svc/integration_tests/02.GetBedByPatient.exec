#!/bin/bash

echo "Patient Exists and assigned:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"patient_id": "8763d32d-73d1-4e53-b367-b970ee5f6c18"}' \
	localhost:3001 \
	services.task_svc.v1.BedService/GetBedByPatient

echo "Patient exists but not assigned to bed:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"patient_id": "2d977c7a-c0a8-4d06-b474-1824d440d8b4"}' \
	localhost:3001 \
	services.task_svc.v1.BedService/GetBedByPatient


echo "Patient does not exist:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"patient_id": "3d977c7a-c0a8-4d06-b474-1824d440d8b4"}' \
	localhost:3001 \
	services.task_svc.v1.BedService/GetBedByPatient

