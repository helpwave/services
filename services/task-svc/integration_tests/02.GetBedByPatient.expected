Patient Exists and assigned:
{
  "room": {
    "id": "c7e6c19e-df2e-4130-b8f3-97b551f911ed",
    "name": "Research and Development",
    "wardId": "51aca99b-2372-40b4-99ed-355dad6c014d"
  },
  "bed": {
    "id": "9f5d0d44-066e-4520-aec3-46a51090d15a",
    "name": "Red"
  }
}
Patient exists but not assigned to bed:
{}
Patient does not exist:
{}
